app/Http/Controllers/Api/V1/EnquiryThreadController.php

```diff
@@ -240,9 +242,51 @@ public function store(Request $request): JsonResponse
 
             $payload['thread']['dataCustodians'] = [];
             $dataCustodians = [];
-            foreach ($payload['thread']['datasets'] as $d) {
-                $t = Team::where('id', $d['team_id'])->first();
-                $dataCustodians[] = $t->name;
+
+
+            if (Feature::active('features.SDEConciergeServiceEnquiry')) {
+                list($conciergeId, $conciergeName) = $this->getNetworkConcierge();
+                $sdeNetwork = DataProviderColl::where('name', 'LIKE', '%SDE%')
+                    ->with('teams')
+                    ->first();
+                $sdeTeamIds = $sdeNetwork ? array_column($sdeNetwork['teams']->toArray(), 'id') : [];
+
+                if ($input['is_general_enquiry']) {
+                    foreach ($payload['thread']['datasets'] as $d) {
+                        $team = Team::where('id', $d['team_id'])->first();
+                        if (in_array($team->id, $sdeTeamIds) && (count($payload['thread']['datasets']) > 1)) {
+                            $teamIds[] = $conciergeId;
+                            $teamNames[] = $conciergeName;
+                        } else {
+                            $teamIds[] = $team->id;
+                            $teamNames[] = $team->name;
+                        }
+                    }
+                } elseif (($input['is_feasibility_enquiry']) || ($input['is_dar_dialogue'])) {
+                    foreach ($payload['thread']['datasets'] as $d) {
+                        $dataset = Dataset::findOrFail($d['dataset_id']);
+                        $metadata = $dataset->latestMetadata()->first();
+                        $gatewayId = $metadata->metadata['metadata']['summary']['publisher']['gatewayId'];
+                        if (is_numeric($gatewayId)) {
+                            $team = Team::where('id', $gatewayId)->first();
+                        } else {
+                            $team = Team::where('pid', $gatewayId)->first();
+                        }
+                        if (in_array($team->id, $sdeTeamIds) && (count($payload['thread']['datasets']) > 1)) {
+                            $teamIds[] = $conciergeId;
+                            $teamNames[] = $conciergeName;
+                        } else {
+                            $teamIds[] = $team->id;
+                            $teamNames[] = $team->name;
+                        }
+                    }
+                }
+            } else {
+
+                foreach ($payload['thread']['datasets'] as $d) {
+                    $t = Team::where('id', $d['team_id'])->first();
+                    $dataCustodians[] = $t->name;
+                }
             }
             $payload['thread']['dataCustodians'] = array_unique($dataCustodians);
```

Comment:
I think lines 295-301 also need to go inside the `else`.  Otherwise the `teamIds` and `teamNames` from the `(Feature::active('features.SDEConciergeServiceEnquiry')) === true` case will be overwritten.